hooks/browser_integration.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook that provides utilities used by Toolkit's browser integration.
+"""
+
+import os
+import hashlib
+import json
+import fnmatch
+import datetime
+
+import sgtk
+
+class BrowserIntegration(sgtk.Hook):
+    def filter_engine_commands(self, commands):
+        """
+        Filters out engine commands that are not desired in the web app menus.
+
+        :param list commands: The list of engine commands to filter.
+
+        :returns: A filtered list of engine commands.
+        :rtype: list
+        """
+        # Filter out any commands that didn't come from an app. This will
+        # filter out things like the "Reload and Restart" command.
+        filtered = dict()
+
+        for name, command in commands.iteritems():
+            if command["properties"].get("app") is not None:
+                filtered[name] = command
+            else:
+                sgtk.platform.current_engine().logger.debug(
+                    "Command filtered out for browser integration: %s" % command
+                )
+        return filtered
+
+    def get_cache_lookup_hash(self, entity_type, pc_descriptor):
```

Comment:
Yeah, next step is to pass up info to SG to configure what data is passed down when messages are sent from the client. This was a much quicker implementation to get the flow up and running as a first step.